src/lnp/application/payment/types.rs

```diff
@@ -265,47 +265,36 @@ impl DumbDefault for TempChannelId {
 #[derive(Wrapper, Clone, Debug, From, PartialEq, Eq)]
 pub struct NodeColor([u8; 3]);
 
-impl LightningEncode for NodeColor {
-    fn lightning_encode<E: io::Write>(
+impl StrictEncode for NodeColor {
+    fn strict_encode<E: io::Write>(
         &self,
         mut e: E,
-    ) -> Result<usize, io::Error> {
+    ) -> Result<usize, StrictError> {
         let len = e.write(self.as_inner())?;
         Ok(len)
```

Comment:
We need just one line: `e.write(self.as_inner())` and that's all

Reply:
This is not working because we also need to convert the error type, which is happening automatically with error propagation. Can you suggest how to do that without error propagation?